### PR TITLE
WIP: Add Markdown Plugin

### LINKED ISSUE #3357: Parcel 2: Markdown transformer
Create the `@parcel/transformer-markdown` package in `packages/transforms/markdown`. This package is a Parcel 2 transform plugin for markdown. It should be based on the `MarkdownAsset` from Parcel 1.

The `@parcel/transformer-markdown` package is responsible for the following things:

* Compiling markdown to HTML using `marked`

### PR DESCRIPTION
**Pull request by JosephMart**
# ↪️ Pull Request

This PR tries to add markdown plugin for transforms. Fixes #3357

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

### CODE REVIEW

File: packages/core/integration-tests/test/markdown.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
+        name: 'index.html',
+        assets: ['index.md']
+      },
+      {
+        type: 'png',
+        assets: ['100x100.png']
+      }
+    ]);
 
     let files = await outputFS.readdir(path.join(__dirname, '/dist'));
```
Comment (JosephMart) [inline packages/core/integration-tests/test/markdown.js]: This test fails at this line with 
```shell
FSError: ENOENT: /..../parcel/packages/core/integration-tests/test/dist does not exist
```
 Where exactly is the output supposed to be? Is the the Markdown Plugin I converted not configured properly? Thoughts?

Comment (devongovett) [reply packages/core/integration-tests/test/markdown.js]: You should use the `distDir` variable from `@parcel/test-utils` here instead.

File: packages/core/integration-tests/test/markdown.js [hunk 2]
```diff
@@ -1,24 +1,23 @@
 const assert = require('assert');
 const path = require('path');
-const {bundle, assertBundleTree, outputFS} = require('@parcel/test-utils');
+const {bundle, assertBundles, outputFS} = require('@parcel/test-utils');
 
-describe.skip('markdown', function() {
-  it('should support bundling Markdown', async function() {
+describe('markdown', function() {
+  it.only('should support bundling Markdown', async function() {
```
Comment (kwelch) [inline packages/core/integration-tests/test/markdown.js]: One thing I have also found helpful is using the `--grep` feature in mocha to regex to the specific test I want to run.

**Comment by Taym95** [comment]
This PR seems inactive for a while, I can create another PR with the changes requested?!

**Comment by mischnic** [comment]
I think the main blocker for a Markdown plugin is what the output format should be (html, raw, a react component), related to that: https://github.com/parcel-bundler/parcel/issues/3477

**Comment by DeMoorJasper** [comment]
Closing in favor of the more complete #3936

**Pull request closed without merging**